internal/tpm/tpm.go

```diff
@@ -1,20 +1,54 @@
 package tpm
 
 import (
+	"bytes"
 	"encoding/hex"
 	"fmt"
 	"io"
+	"os"
 
+	"github.com/google/go-tpm-tools/client"
+	pbattest "github.com/google/go-tpm-tools/proto/attest"
+	pbtpm "github.com/google/go-tpm-tools/proto/tpm"
 	"github.com/google/go-tpm-tools/simulator"
 	"github.com/google/go-tpm/legacy/tpm2"
 	"github.com/google/go-tpm/tpmutil"
 )
 
+const (
+	MinNonceLength     = 8
+	TpmSystemPath      = "/dev/tpm/tpm0"
+	TpmVersionInfoPath = "/sys/class/tpm/tpm0/tpm_version_major"
+)
+
 type TPM struct {
 	devicePath string
 	channel    io.ReadWriteCloser
 }
 
+// Note: this may be a hardware TPM or a software or emulated TPM available to the system
+func TpmExists() bool {
+	if _, err := os.Stat(TpmSystemPath); err == nil {
+		return true
+	}
+	return false
+}
+
+func TpmIsVersion2() error {
```

Comment:
`Is` naming is generally used for a bool return maybe
```suggestion
func ValidateTpmVersion2() error {
```

Reply:
That makes sense